source/docs/advanced/theme/route-dispatcher.md

```diff
@@ -22,103 +22,118 @@ If you come from a Magento background, this is the concept behind [URL Rewrites]
++  if(match.__typename) {
++    // a match have been already loaded (you can still override it or return it as is like we do here)
++    return match;
++  }
++  const myMatch = await loader.loadByUrl(url);
++  if(!myMatch) {
++    return match; // return original match if none is found (this is needed!)
++  }
++  return { // return your custom routable entity
++    path: <canonical_url>,      // depending on your use case can be same as url
++    url,
++    __typename: 'MyCustomType', // the GraphQL type of your custom type (should be added in schema.gql),
++    ...myMatch, // or some adaptation of it. this will be available to your component defined bellow on a prop called `matched`
++  }
++};
++
++export default myCustomUrlMatcher;
+```
 
-3. Override the `route` query in order to use your brand new Loader.
-```js
-const resolvers = {
-  Query: {
-    route: (_, { url }, { loaders }) => {
-      return loaders.MyModuleUrl.matchBy(url)
-        .then((result) => {
-          if (result) {
-            return result;
-          } else {
-            return loaders.Url.matchBy(url);
-          }
-        })
-    },
-  }
+```diff
+# schema.gql
++MyCustomType {
++  title: String! # example field
++  ... # define own schema
++}
+```
+
+```diff
+// loader.js
+const MyCustomLoader = (axiosInstance) => {
+  return {
++    loadByUrl: (url) => {
++      return ( // example: get data by issuing an external api call
++        axiosInstance.get(...)
++        .then((response) => response.data)
++        .catch((err) => err.response?.status === 404 ? null : Promise.reject(err));
++      );
++    },
+  };
 };
 ```
-    Note that we don't want to forget to call the initial `loaders.Url.matchBy` here. If we don't, we will break the base functionalities of Front-Commerce.
 
-    Additionally, since we are using an existing loader, we must make sure that our dependencies are correctly configured in our module. In case of a store using Magento, we should add the dependency on `Magento2/Url` module. See [Graphql module definition Reference](/docs/reference/graphql-module-definition.html#dependencies-optional) for more details.
+    <blockquote class="note">
+    See [Loaders](/docs/advanced/graphql/slim-down-resolvers-with-loaders.html) documentation to learn how to instantiate your loader and add it to your GraphQL context.
+    </blockquote>
 
-Once you've done these three steps, you should be able to test that everything works as expected by using the GraphQL Playground at http://localhost:4000/playground and executing the following query:
+Once you've done, you should be able to test that everything works as expected by using the GraphQL Playground at http://localhost:4000/playground and executing the following query:
 
 ```gql
 {
-  route(url: "my-dynamic-custom-url") {
+  route(url: "some-dynamic-custom-url") {
     url
-    type
-    identifier
-    target_path
-    redirect_type
+    title # example
+    ...   # custom type fields
```

Comment:
```suggestion
```
not needed as the `title` should be sufficient. It makes the example executable without syntax error.